Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -789,37 +793,49 @@ public enum RestError: Error, CustomStringConvertible {
 
 // URL Session extension
 extension RestRequest: URLSessionDelegate {
-
+    
     /// URL session function to allow trusting certain URLs
     public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
         let method = challenge.protectionSpace.authenticationMethod
         let host = challenge.protectionSpace.host
-
+        
         guard let url = URLComponents(string: self.url), let baseHost = url.host else {
             completionHandler(.performDefaultHandling, nil)
             return
         }
-
+        
         let warning = "Attempting to establish a secure connection; This is only supported by macOS 10.6 or higher. Resorting to default handling."
-
+        
         switch (method, host) {
         case (NSURLAuthenticationMethodServerTrust, baseHost):
             #if !os(Linux)
-            guard #available(iOS 3.0, macOS 10.6, *), let trust = challenge.protectionSpace.serverTrust else {
+                guard #available(iOS 3.0, macOS 10.6, *), let trust = challenge.protectionSpace.serverTrust else {
+                    Log.warning(warning)
+                    fallthrough
+                }
+                if let certificateData = NSData(contentsOfFile: Bundle.main.path(forResource: self.pinnedCertificateName, ofType: "der") ?? "") {
```

Comment:
Why are we creating the NSData even if self.pinnedCertificateName is nil? Shouldn't we skip this whole section if it's nil?

Reply:
Added a guard to ensure it's not nil.